java/arcs/core/analysis/DependencyNode.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.analysis
+
+import arcs.core.data.AccessPath
+import arcs.core.data.expression.Expression
+
+/** Field [Identifier]. */
+private typealias Identifier = String
+
+/** Lists of [Identifier]s imply an [AccessPath]. */
+private typealias Path = List<Identifier>
+
+/**
+ * [DependencyNode]s represent how inputs of a Paxel [Expression] contribute to its outputs.
+ *
+ * A [DependencyNode] makes up a Directed-Acyclic-Graph to map fields of an output handle
+ * connection to fields in other handle connections.
+ *
+ * - [DependencyNode.Input] represents an input handle connection and access path.
+ * - [DependencyNode.DerivedFrom] indicates that an input has been modified in the Paxel expression.
+ * - [DependencyNode.AggregateValue] connects partial access paths to other nodes in the graph.
```

Comment:
Since `PrimitiveValue` has been changed to input, may I change back `AggregateValue` to `Associate`?

Reply:
@bgogul What do you think would be the clearest set of names?

Reply:
Let us call it `AssociationNode`?

Reply:
I'm happy with that. :)